src/DotNetLightning.Core/Crypto/Keys.fs

```diff
@@ -0,0 +1,662 @@
+namespace DotNetLightning.Crypto
+
+open System
+open NBitcoin
+open NBitcoin.Crypto
+
+open DotNetLightning.Utils
+open DotNetLightning.Core.Utils.Extensions
+
+// FIXME: Should the [<Struct>]-annotated types here be changed to records or single-constructor discriminated unions? 
+    
+[<AutoOpen>]
+module NBitcoinArithmethicExtensions =
+    /// The functions in this module may fail but it does not return Result and just throw Exception in
+    /// case of failure. Why? because the error case is very, very unlikely to happen.
+    /// (e.g. two keys we operate were complement of each other.)
+    /// It is interesting to consider if it is possible for the attacker to intentionally make this happen,
+    /// but it is way beyond my skills, so we just ignore for now.
+
+    let Secp256k1 = CryptoUtils.impl.newSecp256k1()
+
+    type Key with
+        static member Mul(lhs: Key, rhs: Key): Key =
+            let lhsBytes = lhs.ToBytes()
+            let rhsBytes = rhs.ToBytes()
+            let tweak = ReadOnlySpan(lhsBytes)
+            match Secp256k1.PrivateKeyTweakMultiply(tweak, rhsBytes.AsSpan()) with
+            | true -> Key rhsBytes
```

Comment:
should be new Key(rhsBytes) to avoid warning?

Reply:
Fixed.